// Copyright 2018 Google Inc.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     https://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

// This service implements a simple guestbook
service Guestbook {
    // Add a new post to the guestbook
    rpc AddPost (AddPostRequest) returns (AddPostResponse) {
    }

    // Update an existing post
    rpc UpdatePost (UpdatePostRequest) returns (UpdatePostResponse) {
    }

    // List existing posts
    rpc ListPosts (ListPostsRequest) returns (ListPostsResponse) {
    }
}

message AddPostRequest {
    string title = 1;
    string body = 2;
}

message AddPostResponse {
    string id = 1;
}


message UpdatePostRequest {
    string id = 1;
    string title = 2;
    string body = 3;
}

message UpdatePostResponse {
}


message ListPostsRequest {
    int32 page = 1;
    int32 page_size = 2;
}

message ListPostsResponse {
    repeated Post posts = 1;
}

message Post {
    string id = 1;
    string title = 2;
    string body = 3;
}
